packages/application-generic/src/usecases/get-subscriber-template-preference/get-subscriber-template-preference.usecase.ts

```diff
@@ -153,12 +157,74 @@ export class GetSubscriberTemplatePreference {
   }
 }
 
-function filterActiveChannels(
+function updateOverrideReasons(
+  channelName,
+  sourceName: string,
+  index: number,
+  overrideReasons: IPreferenceOverride[]
+) {
+  const currOverride: IPreferenceOverride = {
+    channel: channelName as ChannelTypeEnum,
+    source: sourceName as 'template' | 'subscriber',
+  };
+
+  const notFoundFlag = -1;
+  const existsInOverrideReasons = index !== notFoundFlag;
+  if (existsInOverrideReasons) {
+    overrideReasons[index] = currOverride;
+  } else {
+    overrideReasons.push(currOverride);
+  }
+}
+
+export function determineOverrides(
+  preferenceSources: Record<'template' | 'subscriber', IPreferenceChannels>
+) {
+  const priorityOrder = ['template', 'subscriber'];
```

Comment:
first, we override by template preference and then by the subscriber as the subscriber is on higher priority.